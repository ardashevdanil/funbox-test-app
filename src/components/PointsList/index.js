import PropTypes from 'prop-types';
import React from 'react';
import { DragDropContext, Droppable } from 'react-beautiful-dnd';
import { connect } from 'react-redux';

import { deletePoint, dragPoint } from '../../redux/modules/points';
import Point from '../Point';

const PointsList = ({
  deleteAction,
  dragAction,
  points,
}) => (
  <DragDropContext

    // FIX: doesn't update state while dragging
    onDragEnd={result => {
      if (!result.destination) return;
      dragAction({
        startIndex: result.source.index,
        endIndex: result.destination.index,
      });
    }}
  >
    <Droppable droppableId='droppable'>
      {provided => (
        <div
          ref={provided.innerRef}
        >
          {points.map((item, index) => (
            <Point
              deleteAction={deleteAction}
              draggableId={item.key}
              index={index}
              key={item.key}
              value={item.value}
            />
          ))}
          {provided.placeholder}
        </div>
      )}
    </Droppable>
  </DragDropContext>
);

const enhance = connect(
  state => ({ points: state.points }),
  { deleteAction: deletePoint, dragAction: dragPoint },
);

export default enhance(PointsList);

PointsList.propTypes = {
  deleteAction: PropTypes.func.isRequired,
  dragAction: PropTypes.func.isRequired,
  points: PropTypes.arrayOf(
    PropTypes.shape({
      coords: PropTypes.array,
      key: PropTypes.number,
      value: PropTypes.string,
    }),
  ).isRequired,
}
