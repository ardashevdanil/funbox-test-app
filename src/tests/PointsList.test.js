import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import PointsList from '../components/PointsList';

Enzyme.configure({ adapter: new Adapter() });

describe('test PointsList', () => {
	const component = Enzyme.shallow(
    <PointsList 
      points={[
        {
          position: 1,
          value: 1,
          key: 1,
        },
        {
          position: 2,
          value: 2,
          key: 2
        }
      ]}
      onDeleteClick={() => null}
      setPointPosition={() => null}
    />
  );

  test('renders correctly', () => {
    expect(component.debug()).toMatchSnapshot();
  });

  test('renders points correctly', () => {
    expect(component.find('Point').length).toBe(2);
  });
});