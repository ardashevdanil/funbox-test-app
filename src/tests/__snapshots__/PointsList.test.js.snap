// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PointsList renders correctly 1`] = `
"<div onScroll={[Function: bound handleScroll]} className=\\"points-list\\">
  <Point position={1} value={1} id={0} onDeleteClick={[Function: onDeleteClick]} scrollTop={0} setPointPosition={[Function: setPointPosition]} />
  <Point position={2} value={2} id={1} onDeleteClick={[Function: onDeleteClick]} scrollTop={0} setPointPosition={[Function: setPointPosition]} />
</div>"
`;
